### PR TITLE
Update to Julia 0.5 to avoid warnings

### PR DESCRIPTION
**Pull request by andreasnoack**

### CODE REVIEW

**Comment by tkelman** [comment]
check existing pr's before you open a new one https://github.com/JuliaLang/BinDeps.jl/pull/229

File: src/BinDeps.jl [hunk 1]
```diff
@@ -415,22 +423,32 @@ module BinDeps
...
-            begin
-                ChangeDirectory(s.builddir)
-                @unix_only FileRule(isempty(s.config_status_dir)?"config.status":joinpath(s.config_status_dir,"config.status"), setenv(`$(s.src)/configure $(s.configure_options) --prefix=$(prefix)`,env))
-                FileRule(s.libtarget,MakeTargets(;env=s.env))
-                MakeTargets("install";env=env)
+        if is_unix() @dependent_steps begin
+                CreateDirectory(s.builddir)
+                begin
+                    ChangeDirectory(s.builddir)
+                    if is_unix()
```
Comment (tkelman) [inline src/BinDeps.jl]: redundant

File: src/BinDeps.jl [hunk 2]
```diff
@@ -415,22 +423,32 @@ module BinDeps
...
-    	@windows_only @dependent_steps begin
-    		begin
-                ChangeDirectory(s.src)
-    			@windows_only FileRule(isempty(s.config_status_dir)?"config.status":joinpath(s.config_status_dir,"config.status"),setenv(`sh -c $cmdstring`,env))
-                FileRule(s.libtarget,MakeTargets())
-                MakeTargets("install")
+    	if is_windows() @dependent_steps begin
+    		  begin
+                    ChangeDirectory(s.src)
+    		  	if is_windows()
```
Comment (tkelman) [inline src/BinDeps.jl]: redundant

File: src/dependencies.jl [hunk 3]
```diff
@@ -468,8 +473,11 @@ function generate_steps(dep::LibraryDependency, h::Autotools,  provider_opts)
     env = Dict{String,String}()
     env["PKG_CONFIG_PATH"] = join(opts[:pkg_config_dirs],":")
     delete!(opts,:pkg_config_dirs)
-    @unix_only env["PATH"] = bindir(dep)*":"*ENV["PATH"]
-    @windows_only env["PATH"] = bindir(dep)*";"*ENV["PATH"]
+    if is_unix()
+        env["PATH"] = bindir(dep)*":"*ENV["PATH"]
+    else is_windows()
```
Comment (tkelman) [inline src/dependencies.jl]: elseif

File: src/dependencies.jl [hunk 4]
```diff
@@ -603,19 +613,20 @@ function check_system_handle!(ret,dep,handle)
...
+elseif is_windows()
     defaults = [Binaries,PackageManager,SystemPaths]
 else
     defaults = [SystemPaths,BuildProcess]
 end
 
 function applicable(dep::LibraryDependency)
     if haskey(dep.properties,:os)
-        if (dep.properties[:os] != OS_NAME && dep.properties[:os] != :Unix) || (dep.properties[:os] == :Unix && !Base.is_unix(OS_NAME))
+        if (dep.properties[:os] != Compat.KERNEL && dep.properties[:os] != :Unix) ||
```
Comment (tkelman) [inline src/dependencies.jl]: Compat.KERNEL is not the right thing to compare against for BinDeps backwards compatibility

Comment (andreasnoack) [inline src/dependencies.jl]: What is the right thing?

**Comment by andreasnoack** [comment]
Probably an easy way to save time.

File: src/dependencies.jl (hunk 4 above)
Comment (tkelman) [inline src/dependencies.jl]: what the deprecation does, preserve the old values

**Pull request closed without merging**